### ISSUE TITLE
Accept async return from language files

### ISSUE THREAD

**Issue by ezypeeze** [nuxt-community/nuxt-i18n#60]
### What problem does this feature solve?

At the moment its possible to load language files asynchronous using dynamic imports, would be also nice to accept promises as returned value.

This way we can fetch translation messages from an API for example.

Example:
```javascript
// <lang_dir>/en-US.js
 module.exports = async function() {
     return i18n.getTranslations('en-US'); // returns a promise
};
```

Also, it would be nice to have a translation message provider option (we could use ```loadLanguagesAsync``` to also accept a function):
```javascript
// nuxt.config.js
{
    modules: [
        ['nuxt-i18n', {
            loadLanguagesAsync: function(langCode) {
                return i18n.getTranslations(langCode); // returns a promise
            }
        }]
    ]
}
```

<div align="right"><sub><em>This feature request is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt-community/nuxt-i18n/issues/c50">#c50</a>)</em></sub></div>

**Comment by michaelwills** [comment]
We can already return a promise in the language files but it would be neat to do so in `loadLanguagesAsync` or some other property. Looks like some updates to the templates would be needed.

It's helpful as right now each language needs to duplicate the same albeit small template to do what could be done within the config.

**Comment by ezypeeze** [comment]
@michaelwills 

Something like this would do the job.
```javascript
export function loadLanguageAsync (i18n, lang) {
  if (!i18n.loadedLanguages) {
    i18n.loadedLanguages = []
  }
  if (!i18n.loadedLanguages.includes(lang)) {
    const langOptions = i18n.locales.find(l => l.code === lang)
    if (langOptions) {
      const { langFile } = langOptions;
      return import(/* webpackChunkName: "lang-[request]" */ '~/<%= options.langDir %>'+langFile)
        .then(result => result instanceof Function ? result() : result) // check functions and call them
        .then(msgs => {
          i18n.setLocaleMessage(lang, msgs)
          i18n.loadedLanguages.push(lang)
          return
        })
    }
  }
  return Promise.resolve()
}
```

**Comment by michaelwills** [comment]
@ezypeeze something like that yes. I haven't checked to compare but `result()` might need to pass the `lang` parameter but just looking at this it looks correct. Nice!

**Comment by paulgv** [comment]
Closed by #91
